packages/content-conformance/README.md

```diff
@@ -126,6 +124,78 @@ export default {
   - `contentFile` (function) Called when a `ContentFile` is visited.
   - `dataFile` (function) Called when a `DataFile` is visited.
 
+### Writing an executor
+
+A rule executor contains the logic for the conformance rule. Each executor type is run once for each file of the provided `type` (`content`, or `data`), or once for the entire project if the type is `structure`.
+
+Each executor has access to two arguments:
+
+- `file` (`ContentFile` | `DataFile`) The file being checked by the rule.
+- `context` (object) An object containing global information and a `report()` method (see `ConformanceRuleContext` in [types.ts](./src/types.ts)).
+
+#### `context.report()`
+
+An executor should contain logic related to the rule to determine if the given file is in violation. If a violation is detected, the executor should call `context.report()`:
+
+```js
+{
+  async contentFile(file, context) {
+    if (hasViolation) {
+      context.report('This rule was broken', file)
+    }
+  }
+}
+```
+
+The `context.report()` method accepts three arguments:
+
+- `message` (string) The message that will be used in generated output.
+- `file` (object) The file that the report is associated with.
+- `node` (object) The node that the violation is associated with. Allows positional information to be displayed in the report. (optional)
+
+#### `ContentFile` visitors
+
+When defining a `contentFile` executor, the `file` argument exposes a `visit()` method. This allows the rule to interact with the Abstract Syntax Tree (AST) generated from the source file. It allows checks to be written against specific parts of a content file with minimal boilerplate or extra processing. Usage of `file.visit()` is very similar to [`unist-util-visit`](https://www.npmjs.com/package/unist-util-visit).
+
+```js
+{
+  async contentFile(file, context) {
+    file.visit(['link'], node => {
+      const hasViolation = node.url.includes('example.com')
+      if (hasViolation) {
+        context.report(`Link to example.com detected: ${node.url}`, file, node)
+      }
+    })
+  }
+}
+```
```

Comment:
💭 A quick note here about how "level" is decoupled & set at the config level (`### Configuring rules`) could be a helpful breadcrumb for readers who like to bounce around